src/neo-vm/Unsafe.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace Neo.VM
+{
+    internal static class Unsafe
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        unsafe public static bool Equals(byte[] x, byte[] y)
+        {
+            if (ReferenceEquals(x, y)) return true;
+            if (x is null || y is null) return false;
+            int len = x.Length;
+            if (len != y.Length) return false;
+            fixed (byte* xp = x, yp = y)
+            {
+                long* xlp = (long*)xp, ylp = (long*)yp;
+                for (; len >= 8; len -= 8)
+                {
+                    if (*xlp != *ylp) return false;
+                    xlp++;
+                    ylp++;
+                }
+                byte* xbp = (byte*)xlp, ybp = (byte*)ylp;
+                for (; len > 0; len--)
+                {
+                    if (*xbp != *ybp) return false;
+                    xbp++;
+                    ybp++;
+                }
+            }
+            return true;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        unsafe public static void MemoryCopy(byte[] src, int srcOffset, byte[] dst, int dstOffset, int count)
+        {
+            fixed (byte* sp = &src[srcOffset], dp = &dst[dstOffset])
+            {
+                Buffer.MemoryCopy(sp, dp, dst.Length - dstOffset, count);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        unsafe public static bool NotZero(byte[] x)
+        {
+            if (x is null)
+                throw new ArgumentNullException(nameof(x));
+            int len = x.Length;
+            if (len == 0) return false;
+            fixed (byte* xp = x)
+            {
+                long* xlp = (long*)xp;
+                for (; len >= 8; len -= 8)
+                {
+                    if (*xlp != 0) return true;
+                    xlp++;
+                }
+                byte* xbp = (byte*)xlp;
+                for (; len > 0; len--)
+                {
+                    if (*xbp != 0) return true;
+                    xbp++;
+                }
+            }
+            return false;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        unsafe public static int ToInt32(byte[] value, int startIndex)
+        {
+            fixed (byte* pbyte = &value[startIndex])
```

Comment:
We need to check if the length is more than 4 bytes, otherwise we are taking random memory

Reply:
This is `unsafe` method. And we currently only use it in `HashComparer`, and `HashComparer` will only accept 20-byte array. So we don't need to check it.